src/web/app/pages-session/session-submission-page/session-submission-page.component.spec.ts

```diff
@@ -1,24 +1,616 @@
 import { AjaxLoadingModule } from '../../components/ajax-loading/ajax-loading.module';
+import { CommentRowModel } from '../../components/comment-box/comment-row/comment-row.component';
 import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
 import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
+import {
+  FeedbackResponseRecipientSubmissionFormModel,
+  QuestionSubmissionFormModel,
+} from '../../components/question-submission-form/question-submission-form-model';
 import {
   QuestionSubmissionFormModule,
 } from '../../components/question-submission-form/question-submission-form.module';
 import { TeammatesCommonModule } from '../../components/teammates-common/teammates-common.module';
-import { SessionSubmissionPageComponent } from './session-submission-page.component';
+import { SavingCompleteModalComponent } from './saving-complete-modal/saving-complete-modal.component';
+import { FeedbackResponsesResponse, SessionSubmissionPageComponent } from './session-submission-page.component';
+import Spy = jasmine.Spy;
 
 describe('SessionSubmissionPageComponent', () => {
+  const deepCopy: any = (obj: any) => JSON.parse(JSON.stringify(obj));
+
+  const testStudent: Student = {
+    email: 'alice@tmms.com',
+    courseId: 'course-id',
+    name: 'Alice Betsy',
+    teamName: 'Team 1',
+    sectionName: 'Section 1',
+  };
+
+  const testInstructor: Instructor = {
+    courseId: 'course-id',
+    email: 'test@example.com',
+    name: 'Instructor Ho',
+    joinState: JoinState.JOINED,
+  };
+
+  const testOpenFeedbackSession: FeedbackSession = {
+    feedbackSessionName: 'First Session',
+    courseId: 'CS1231',
+    timeZone: 'Asia/Singapore',
+    instructions: 'Instructions',
+    submissionStartTimestamp: 1000000000000,
+    submissionEndTimestamp: 1500000000000,
+    gracePeriod: 0,
+    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
+    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+    publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
+    isClosingEmailEnabled: true,
+    isPublishedEmailEnabled: true,
+    createdAtTimestamp: 0,
+  };
+
+  const testClosedFeedbackSession: FeedbackSession = {
+    feedbackSessionName: 'Second Session',
+    courseId: 'CFG1010',
+    timeZone: 'Asia/Singapore',
+    instructions: 'Roots and Shoots',
+    submissionStartTimestamp: 1500000000000,
+    submissionEndTimestamp: 2000000000000,
+    gracePeriod: 0,
+    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
+    submissionStatus: FeedbackSessionSubmissionStatus.CLOSED,
+    publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
+    isClosingEmailEnabled: true,
+    isPublishedEmailEnabled: true,
+    createdAtTimestamp: 0,
+  };
+
+  const testVisibleNotOpenFeedbackSession: FeedbackSession = {
```

Comment:
Instead of having 3 separate sessions, can we use 1 and modify its attrs based on what we need e.g set visibility corresponding to the test case